src/Microsoft.Fast.Components.FluentUI/Components/NavMenu/FluentNavMenuGroup.razor.cs

```diff
@@ -108,38 +112,42 @@ public FluentNavMenuGroup()
         .AddStyle(Style)
         .Build();
 
-    private bool HasCollapsedIcon => !string.IsNullOrWhiteSpace(Icon) || CollapsedIconContent is not null;
-    private bool HasExpandedIcon => !string.IsNullOrWhiteSpace(Icon) || ExpandedIconContent is not null;
+    private bool HasIcon => !string.IsNullOrWhiteSpace(Icon) || IconContent is not null;
 
-    protected override void OnParametersSet()
-    {
-        NavMenu.AddNavMenuGroup(this);
+    internal bool HasNavMenuGutterIcon => !string.IsNullOrWhiteSpace(NavMenuGutterIcon) || NavMenuGutterIconContent is not null;
 
-        if (_expanded != Expanded)
-        {
-            _expanded = Expanded;
-            if (OnExpandedChanged.HasDelegate)
-                OnExpandedChanged.InvokeAsync(_expanded);
-        }
+    public override async Task SetParametersAsync(ParameterView parameters)
+    {
+        bool originalExpanded = Expanded;
+        await base.SetParametersAsync(parameters);
+        if (Expanded != originalExpanded)
+            await OnExpandedChanged.InvokeAsync(Expanded);
     }
 
+    protected override void OnInitialized()
+    {
+        base.OnInitialized();
+        NavMenu.AddNavMenuGroup(this);
+    }
 
-
-    /// <summary />
-    internal async Task ExpandMenu()
+    private async Task ExpandMenu()
     {
         if (Disabled)
             return;
 
-        // Expand the Menu Group if the user click on it
+        // Expand the Nav Menu and this Group if the user clicks on its NavMenuToggleIcon
         if (!NavMenuExpanded)
         {
-            Selected = true;
             await NavMenu.CollapsibleClickAsync();
+
+            Expanded = true;
+            await OnExpandedChanged.InvokeAsync(Expanded);
+
+            Selected = true;
```

Comment:
Before: Click icon in collapsed nav menu, nav menu expands, then click to expand the group.

If I want to see what's in the group, I want to expand both the nav menu and the group with a single click.